sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosEndToEndOperationLatencyPolicyConfigBuilder.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.models.CosmosEndToEndOperationLatencyPolicyConfig;
+
+import java.time.Duration;
+
+/**
+ * Builder for CosmosEndToEndOperationLatencyPolicyConfig
+ */
+public class CosmosEndToEndOperationLatencyPolicyConfigBuilder {
+    private final boolean isEnabled;
+    private Duration endToEndOperationTimeout = Duration.ofSeconds(2); //defaults to two seconds
```

Comment:
Default duration of 2seconds OK?

Reply:
yeah, sounds good to me. However, this being default to 2 seconds, I wonder if we should also change the default of networkRequestTimeout in DirectConnectionConfig to 2 seconds or less?
But also, since this is opt-in, may be we don't, just brainstorming, @xinlian12 thoughts?

Reply:
hmm, still think 5s should be the default network request timeout.....

Reply:
I would make the end-to-end timeout required. No reason to choose this retry strategy without an end-to-end timeout. That is the one required paramter (should be in ctor)

Reply:
If it has to be provided always, there will be no DEFAULT short cut then.
    public static final CosmosEndToEndOperationConfig DEFAULT =
        new CosmosEndToEndOperationConfigBuilder().build();

Reply:
We don't need a DEFAULT shortcut here

Reply:
That also prevent the mutability issue (which means DEFAULT would need to create a new instance every single time anyway because tiemouts and isenabled become mutable eventuall)

Reply:
timeout duration is a required parameter now. So no more defaults